src/components/OrdersWidget/index.tsx

```diff
@@ -0,0 +1,226 @@
+import React, { useMemo } from 'react'
+import { Link } from 'react-router-dom'
+import styled from 'styled-components'
+import { faExchangeAlt, faChartLine, faTrashAlt, faExclamationTriangle } from '@fortawesome/free-solid-svg-icons'
+import { FontAwesomeIcon } from '@fortawesome/react-fontawesome'
+
+import Widget from 'components/layout/Widget'
+import Highlight from 'components/Highlight'
+import { tokenListApi } from 'api'
+import { getToken } from 'utils'
+import { Network, TokenDetails } from 'types'
+import OrderRow from './OrderRow'
+
+const OrdersWrapper = styled(Widget)`
+  > a {
+    margin-bottom: -2em;
+  }
+`
+
+const ButtonWithIcon = styled.button`
+  min-width: 10em;
+
+  > svg {
+    margin: 0 0.25em;
+  }
+`
+
+const CreateButtons = styled.div`
+  margin-top: 2em;
+  display: flex;
+  justify-content: center;
+  align-items: flex-start;
+
+  .strategy {
+    display: flex;
+    flex-direction: column;
+    justify-content: flex-start;
+    align-items: flex-start;
+
+    a,
+    button {
+      margin-left: 0.5em;
+    }
+  }
+`
+
+const OrdersForm = styled.div`
+  margin-top: 2em;
+  margin-left: 2em;
+
+  .infoContainer {
+    display: grid;
+    grid-template-columns: repeat(2, 1fr);
+
+    .warning {
+      justify-self: end;
+    }
+  }
+
+  .ordersContainer {
+    // negative left margin to better position "hidden" elements
+    margin: 2em 0 2em -3em;
+
+    display: grid;
+    // 6 columns:
+    // loading indicator | select checkbox | order details | unfilled | available | expires
+    grid-template-columns: 3em 4em minmax(20em, 1.5fr) repeat(3, 1fr);
+    grid-row-gap: 1em;
+    place-items: center;
+  }
+
+  .headerRow {
+    // make the contents of this div behave as part of the parent
+    // grid container
+    display: contents;
+
+    text-transform: uppercase;
+    font-weight: bold;
+    font-size: 0.75em;
+
+    .title {
+      // create a divider line only bellow titled columns
+      border-bottom: 2px solid #ededed;
+      // push the border all the way to the bottom and extend it
+      place-self: stretch;
+
+      // align that text!
+      display: flex;
+      align-items: center;
+      justify-content: center;
+      text-align: center;
+    }
+
+    > * {
+      // more space for the divider line
+      padding-bottom: 0.5em;
+    }
+  }
+
+  .orderRow {
+    display: contents;
```

Comment:
Are you using divs with `display: contents` for easier group styling? Otherwise there's not much use to them compared to React.Freagments

Reply:
I prefer this for better grouping and semantics, keeping each row in a separated `<div>`.
Otherwise having all `cells` in a flat `<div>` look like a big mess to me.

Hope that still works with grid-area though.